.github/workflows/add-changeset.yml

```
@@ -0,0 +1,75 @@
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+      # Add an empty changeset (defaults to patch with generic message)
+      - name: Create Changeset File
+        run: pnpm changeset add --empty
+        # No extra env needed here, git config is set globally for the step
+
+      # Commit and push the changeset file back to the PR branch
+      - name: Commit and Push Changeset
+        run: |
```

actionlint (1.7.4)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)